src/roman_datamodels/testing.py

```diff
@@ -109,13 +109,18 @@ def _assert_value_is_copy(value1, value2, deepcopy=False):
         elif isinstance(value1, Model):
             assert value1 is not value2
         else:
-            # Hashable objects cannot be copied as they are immutable.
+            # If not hashable, then can be copied.
             try:
                 hash(value1)
             except TypeError:
                 assert value1 is not value2
             else:
-                assert value1 is value2
+                # Normally hashable objects do not deepcopy.
+                # GWCS violates this.
```

Comment:
Just out of curiosity---I would have said we already had models with GWCS in them (e.g., WfiImage).  What's different here?

Reply:
On my list to investigate. The known difference is that the GWCS objects are usually in `meta.wcs` and not top-level, such as these are. This implies that a different decision path occurs WRT the meta tree.

Regardless, the code itself is based on a false assumption; [see python docs, section 6.10.1](https://docs.python.org/3/reference/expressions.html#value-comparisons). The relevant words are at the end of that section about `hash()` and equality. Python does not enforce/cannot enforce this, since a class can define its own hash. As such, the current code modification is theoretically needed anyways. Regarding GWCS itself, an issue will be created for discussion there.

Reply:
Would you open a roman_datamodels issue or rcal ticket about this?